### ISSUE TITLE
S2 — Two guards the comments promise and the code never wrote [Epic #353]

### ISSUE THREAD

**Issue by Polichinel** [views-platform/views-hydranet#355]
**Epic:** #353 · **Depends on:** none (parallel with S1) · **Blocks:** —

## Background

Two comments in `training_engine.py` assert that a config validator prevents a dangerous
combination. **Neither validator exists.** This is **C-303** — prose asserting a guard the code does
not implement — and the register already carries it thirteen times.

### (a) The pushforward feeds the UNGATED body mean

`training_engine.py:786-793`:

```python
fed = _family_feedback_log1p(
    t1_pred, family, ss_feedback,
    torch.sigmoid(t1_pred_class), forecast_composition, gate_threshold,
).detach()
```

`_family_feedback_log1p:337` dispatches `if mode != "sample": return _family_target_log1p_mean(reg, family)`.
`ss_feedback` defaults to `"mean"` (`config_initializer.py:191`). And `_family_target_log1p_mean:287`
**ignores the gate entirely** — `composition` and `threshold` are not even in its signature.

The correctly-composed helper **exists 480 lines above**: `_family_composed_mean_log1p:302`, whose own
docstring says it must be used instead *"because that one ignores the gate entirely, so it is the
analogue of an UNCOMPOSED draw and would push gradient for a quantity the forward pass never
produced."*

So the auxiliary unroll is fed `log1p(E[Y|body])` while deployment emits `log1p(gate × body)`. With
this model's gate at ~0.003–0.05, the pushforward input is **one to two orders of magnitude larger
than anything inference produces**. The arm runs clean and is scored as the pushforward treatment.

**The guard that would catch it is nested out of reach.** The C-259 check —
*"the training mean feedback is UNGATED while inference's mean is gated"* — is at
`config_initializer.py:1234`, inside `if self.ss_epsilon_max > 0:` at `:1212`. A pushforward arm
never sets `ss_epsilon_max`; the branch gates on `pushforward_weight > 0.0 and family is not None`
(`training_engine.py:781-785`). Verified: a config with `ss_epsilon_max=0.0`, `pushforward_weight=0.1`,
`forecast_composition='soft_gate'` and the default `ss_feedback='mean'` **validates**.

### (b) BPTT-SA on a legacy head

`training_engine.py:578-586`, verbatim tail:

> *"The helper promises `fed` is "the sole tensor through which credit crosses"; that is true for a
> family head, where `fed` is derived, and false here. **A config validator rejects the combination
> outright, so this branch is defence in depth.**"*

Exhaustive grep of `config_initializer.py` for `ss_backprop_through_feedback` returns the field
declaration and **two** validators: `reject_feedback_clip_without_the_wire:806` (rejects the clip
*without* the flag — the opposite direction) and `reject_bptt_sa_that_cannot_run:823` (rejects the
flag when scheduled sampling is inactive). **Neither checks the head type.**

Verified: `HydraNetConfig(output_distribution='standard', ss_schedule='linear', ss_warmup_lessons=10,
ss_epsilon_max=1.0, ss_feedback='mean', ss_backprop_through_feedback=True, ss_feedback_grad_clip=5.0)`
validates. On that path `fed = t1_pred` and `prev_pred = fed` un-detached (`:589`), so the BPTT wire
**is** connected across all ~383 steps while the per-step clip silently no-ops and `fed_grad_max` logs
a constant `0.0` that reads as a healthy gradient — in the regime GRAD-TRAJ measured blowing to
**9.4e9** and overflowing float32 at lesson 48.

`reject_pushforward_without_a_family:844` is exactly this validator for the pushforward feature, and
`reject_feedback_clip_without_the_wire`'s own docstring (`:812`) cites it as the pattern to mirror.

## Work

- [ ] **(a) needs-decision:** compose the pushforward feedback (use `_family_composed_mean_log1p`), or
      reject the combination in config. Composing changes what a pushforward arm trains on — no arm
      has run, `pushforward_weight` defaults to `0.0` and no roster config sets it, so either is safe
      today. **Record the reasoning; do not pick silently.**
- [ ] Implement the chosen option for (a)
- [ ] **(b)** Write the missing validator: reject `ss_backprop_through_feedback=True` on a head with
      no registered family, mirroring `reject_pushforward_without_a_family`
- [ ] Correct or delete the comment at `:578-586` — it must not survive in its current form either way
- [ ] Audit the remaining "a validator rejects this" comments in `views_hydranet/` and pin each to the
      validator that implements it, or remove the claim

## Acceptance criteria

- [ ] A pushforward config with a gated composition and `ss_feedback='mean'` either feeds the composed
      mean or fails to validate — not both, and not neither
- [ ] `ss_backprop_through_feedback=True` on a legacy/point/quantile head raises, naming the head
- [ ] No comment in `views_hydranet/` asserts a validator that grep cannot find — **verified by grep
      output pasted into the PR**, not by reading
- [ ] Tests for both, each **failing when its fix is reverted** (**C-329**)

## Testing / validation

- New cases in `tests/test_falsification_loss_param_validation.py` or alongside the existing
  `reject_pushforward_without_a_family` tests
- `pytest tests/test_bptt_sa.py` — note **S9** repairs a test in this file that always skips in CI
- `ruff check .` and the full suite

## Files

- `views_hydranet/train/training_engine.py` (`:287`, `:302`, `:337`, `:578-593`, `:781-793`)
- `views_hydranet/utils/config_initializer.py` (`:191`, `:806`, `:823`, `:844`, `:1212-1243`)
- `docs/CICs/TrainingEngine.md`, `docs/CICs/HydraNetConfig.md` if a field's contract changes

**Comment by Polichinel** [comment]
## Implemented — both, and the `needs-decision` on (a) is taken: **compose, do not reject**

### (a) The pushforward now feeds the composed mean

`_family_feedback_log1p`'s `'mean'` branch called `_family_target_log1p_mean(reg, family)`, which
ignores the gate entirely. It now calls `_family_composed_mean_log1p(reg, family, gate, composition,
threshold)` — the helper whose own docstring already said it was the right one.

**Why compose rather than reject the combination:** inference's mean path *is* gated
(`_emit_magnitude` composes with the cls gate at emit time, ADR-069). A pushforward that fed the
ungated mean would train on a different exposure than the model deploys under — the C-259 class,
which the nested validator could not see because a pushforward arm never sets `ss_epsilon_max`.
Rejecting would have closed the arm; composing makes it correct.

Tests: `test_the_mean_feedback_is_composed_not_the_raw_body_mean` and
`test_the_surrogate_is_the_COMPOSED_mean_not_the_bare_body_mean` in `tests/test_bptt_sa.py`.
**Deletion test:** point the branch back at `_family_target_log1p_mean` → red.

### (b) The validator the comment promised now exists

`reject_bptt_sa_without_a_family`: `ss_backprop_through_feedback=True` on a head that does not
resolve to a distribution family is rejected, mirroring `reject_pushforward_without_a_family`. The
call-site comment now names the validator it claims. On a legacy head the fed tensor *is* the
regression loss's input, the straight-through estimator does not apply, the per-step clip was
skipped, and `fed_grad_max` logged a constant `0.0` that read as a healthy gradient.

Tests in `tests/test_rollout_horizon_config.py`: rejected on `standard` and `quantile`; constructs on
`nb`; `False` allowed on any head. **Deletion test:** revert → red.

## Register / CIC

Both are **C-303**, taking it to fourteen. `docs/CICs/HydraNetConfig.md` carries the note; field
count unchanged at 99.

**Comment by Polichinel** [comment]
Merged to `development` in PR #372 (`1d6cef8`). Closed by hand — merges to `development` do not auto-close.
